src/Features/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexEmbeddedCompletionProvider.cs

```diff
@@ -0,0 +1,498 @@
+            context.AddIfMissing($"[  {Regex_base_group}  -[  {Regex_excluded_group}  ]", Regex_character_class_subtraction_short, Regex_character_class_subtraction_long, parentOpt, positionOffset: "[".Length, insertionText: "[-[]]");
+        }
+
+        private void ProvideEscapeCategoryCompletions(EmbeddedCompletionContext context)
+        {
+            foreach (var (name, (shortDesc, longDesc)) in RegexCharClass.EscapeCategories)
+            {
+                var displayText = name;
+
+                // There are some internal escape categories the regex engine has (like _xmlI).
+                // Just filter out here so we only show the main documented regex categories.
+                // Note: we still include those in RegexCharClass.EscapeCategories because we
+                // don't want to report an error on code that does use these since the .net
+                // regex engine will allow them.
+                if (displayText.StartsWith("_"))
+                {
+                    continue;
+                }
+
+                var description = longDesc.Length > 0
+                    ? longDesc
+                    : string.Format(Regex_unicode_general_category_0, name);
+
+                context.AddIfMissing(new RegexItem(
+                    displayText, shortDesc, description,
+                    change: CompletionChange.Create(
+                        new TextChange(new TextSpan(context.Position, 0), name), newPosition: null)));
+            }
+        }
+
+        private void ProvideEscapeCompletions(
+            EmbeddedCompletionContext context, bool inCharacterClass, RegexNode parentOpt)
+        {
+            if (parentOpt != null && !(parentOpt is RegexEscapeNode))
+            {
+                return;
+            }
+
+            if (!inCharacterClass)
+            {
+                context.AddIfMissing(@"\A", Regex_start_of_string_only_short, Regex_start_of_string_only_long, parentOpt);
+                context.AddIfMissing(@"\b", Regex_word_boundary_short, Regex_word_boundary_long, parentOpt);
+                context.AddIfMissing(@"\B", Regex_non_word_boundary_short, Regex_non_word_boundary_long, parentOpt);
+                context.AddIfMissing(@"\G", Regex_contiguous_matches_short, Regex_contiguous_matches_long, parentOpt);
+                context.AddIfMissing(@"\z", Regex_end_of_string_only_short, Regex_end_of_string_only_long, parentOpt);
+                context.AddIfMissing(@"\Z", Regex_end_of_string_or_before_ending_newline_short, Regex_end_of_string_or_before_ending_newline_long, parentOpt);
+
+                context.AddIfMissing($@"\k<  {Regex_name_or_number}  >", Regex_named_backreference_short, Regex_named_backreference_long, parentOpt, @"\k<".Length, insertionText: @"\k<>");
+                // context.AddIfMissing(@"\<>", "", "", parentOpt, @"\<".Length));
```

Comment:
I'll put a comment.  `\<>` is considered deprecated and is recommended that you use.  Instead, the recommendation is to use `\k<>`.  the .net regex guide doesn't even mention the former.  But it still supports it for legacy reasons.  So i basically took all the constructs my parser supports and pasted it here.  When i got to this one, i commented it out since we don't even want to recommend it.

Reply:
Ah, OK. Given it looked incomplete it looked more like a copy/paste mistake.

Reply:
added comment!